db/version_set.cc

```diff
@@ -1967,7 +1967,8 @@ void Version::Get(const ReadOptions& read_options, const LookupKey& k,
                   MergeContext* merge_context,
                   SequenceNumber* max_covering_tombstone_seq, bool* value_found,
                   bool* key_exists, SequenceNumber* seq, ReadCallback* callback,
-                  bool* is_blob, bool do_merge) {
+                  bool* is_blob, bool do_merge,
+                  std::shared_ptr<PinnedIteratorsManager> pinned_iters_mgr) {
```

Comment:
Passing shared_ptr in this way can be expensive, as it will increment an atomic counter. We had several performance regressions due to this.

Reply:
Not related to this PR, but the way we do it is that if the caller holds the/an ownership to this object, we just pass the raw pointer down to the child functions. Not sure whether there is going to be a better way of doing it with C++17 though.